tst/testbugfix/2018-09-13-MTC.tst

```diff
@@ -0,0 +1,6 @@
+#MTC as arises when testing the order of M22 as FP group. Timmons' report
+gap> F:=FreeGroup("a","b");;
+gap> rels:=ParseRelators(F,"a2,b4,(ab)11, (ab2)5,[a,bab]3,(ababaB)5");;
+gap> G:=F/rels;;
+gap> Size(G);
+443520
```

Comment:
As for usability, I also added a small change to how cyclic subgroups are found that allows for a smaller index MTC and thus reduces the user wait time by another factor 3.

Reply:
Yeah, it now only takes 51 seconds; also, the computation w/o this PR terminated after 519 minutes.

And yeah, I realize that this is due to the completely new MTC implementation -- and I still think moving to your new MTC implementation was the right choice. And since it is new, I am very hopeful that we can further improve it. Besides algorithmic tweaks and fixes like in this PR (great work), a profile might reveal a few more hotspots; it might be that optimizing those (perhaps also rewriting a few more bits in C) could get us even closer to old times. Or maybe not, my point is, we haven't really tried yet to optimize this further, which makes me quite hopeful that there are opportunities for it left. So I am quite positive about this.

Reply:
@hulpke - would you mind giving a link to the report on the original MTC bug?

Reply:
@dimpase

the report was a forum email (how to link, as we have no official archive?). Also it only specified "Mathieu groups are slow" and did not bother to supply the presentation.
If someone wants to submit it as a formal issue, I'll link to that.

Reply:
@hulpke we do have an official archive, which can be found from the GAP website via [The GAP Forum](https://www.gap-system.org/Contacts/Forum/forum.html) -> [The GAP Forum Archive](https://www.gap-system.org/Doc/forumarchive.html) -> [Archive of postings since December 2003](https://mail.gap-system.org/pipermail/forum/) etc.

The message in question is https://mail.gap-system.org/pipermail/forum/2018/005793.html.